src/libraries/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -669,6 +681,31 @@ private static void ValidateReadArgs(byte[] buffer, int offset, int count)
             public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) { throw new NotSupportedException(); }
             public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default) { throw new NotSupportedException(); }
         }
+
+
+        private static void WriteToStream(Stream stream, string content, Encoding encoding)
+        {
+            if (ReferenceEquals(encoding, Encoding.Latin1) ? content.Length <= 512
+                : (ReferenceEquals(encoding, Encoding.UTF8) && content.Length <= 128))
+            {
+                // Fast-path for the expected common cases
+                Span<byte> buffer = stackalloc byte[512];
+                int written = encoding.GetBytes(content, buffer);
+                stream.Write(buffer.Slice(0, written));
+            }
+            else
+            {
+                stream.Write(encoding.GetBytes(content));
```

Comment:
We're not concerned about the byte[] allocation here because it's expected to be rare?

Reply:
Yes, Latin1 is the default and I am expecting the 99% use case for non-default to be UTF8.
We could special-case Ascii as well if we believe there is a use-case for it.

For the fallback for any custom encoding / long lengths, we could use `GetByteCount` and encode into the ArrayPool instead of allocating.

Reply:
Now using the ArrayPool as a fallback from stackalloc.